lib/livebook_web/live/hooks/sidebar_hook.ex

```diff
@@ -35,4 +45,38 @@ defmodule LivebookWeb.SidebarHook do
   end
 
   defp handle_event(_event, _params, socket), do: {:cont, socket}
+
+  @supervisor Livebook.HubsSupervisor
+  @registry Livebook.HubsRegistry
+
+  defp connect_enterprise_hubs(socket) do
+    for %{provider: %Enterprise{} = enterprise} <- socket.assigns.saved_hubs, reduce: socket do
```

Comment:
You should not connect them here. We will connect them when the application starts. Whenever a hub is saved, we start another one. Whenever a hub is deleted, we stop it. The remaining of the app will work based on the assumption that those processes are always started.

I am wondering if perhaps it is worth extracting this functionality into a separate PR for review?

Reply:
I agree with extracting this functionality into another PR, so we can focus the review